src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnectionFactory.cs

```diff
@@ -39,7 +39,7 @@ override protected DbConnectionInternal CreateConnection(DbConnectionOptions opt
             SqlInternalConnection result = null;
             SessionData recoverySessionData = null;
 
-            SqlConnection sqlOwningConnection = owningConnection as SqlConnection;
+            SqlConnection sqlOwningConnection = (SqlConnection)owningConnection;
```

Comment:
@saurabh500 Whatever, there're two casts like this in the code below this code - they would fail anyway if the object is of wrong type. It's clearer to have a single line of code which fails if the object is of wrong type, this also allows to not have the same cast repeated in the code.